spec/cypress/integration/lead-provider-search-schools.js

```diff
@@ -0,0 +1,126 @@
+describe("Lead provider search schools page", () => {
+  beforeEach(() => {
+    cy.appScenario("schools");
+
+    // Not sure why cypress isn't doing this…
+    sessionStorage.clear();
+  });
+
+  it("should supporting searching for and adding partnerships", () => {
+    cy.login("admin");
+
+    // Unfiltered schools - first page
+    cy.visit("/lead-provider/search-schools");
+
+    cy.contains("49 results found");
+
+    cy.get(".partnerships-submit").as("submit").should("be.disabled");
+    cy.get('[data-cy="partnerships-clear"]').should("not.be.visible");
+
+    cy.get('[type="checkbox"][name="partnership_form[schools][]"]').as(
+      "checkboxes"
+    );
+
+    cy.get("@checkboxes").eq(1).click();
+    cy.get("@checkboxes").eq(3).click();
+    cy.get("@checkboxes").eq(6).click();
+
+    cy.get("@submit")
+      .should("contain", "Add partnerships with 3 schools")
+      .should("not.be.disabled");
+
+    cy.get('[data-cy="partnerships-clear"]').should(
+      "contain",
+      "Remove all 3 schools"
+    );
+
+    cy.reload();
+
+    cy.get(".partnerships-submit")
+      .should("contain", "Add partnerships with 3 schools")
+      .should("not.be.disabled");
+
+    cy.get('[data-cy="partnerships-clear"]').should(
+      "contain",
+      "Remove all 3 schools"
+    );
+
+    cy.get('[type="checkbox"][name="partnership_form[schools][]"]').as(
+      "checkboxes"
+    );
+
+    // @todo replace with a visual test
+    cy.get("@checkboxes").eq(1).should("be.checked");
+    cy.get("@checkboxes").eq(2).should("not.be.checked");
+    cy.get("@checkboxes").eq(3).should("be.checked");
+    cy.get("@checkboxes").eq(4).should("not.be.checked");
+    cy.get("@checkboxes").eq(6).should("be.checked");
+
+    // Unfiltered schools: second page
+    cy.get('[rel="next"]:first').click();
+
+    cy.get('[type="checkbox"][name="partnership_form[schools][]"]').as(
+      "checkboxes"
+    );
+
+    cy.get("@checkboxes").eq(7).should("not.be.checked").click();
+    cy.get("@checkboxes").eq(14).should("not.be.checked").click();
+
+    cy.get(".partnerships-submit")
+      .should("contain", "Add partnerships with 5 schools")
+      .should("not.be.disabled");
+
+    // Filtered schools
+    cy.get('[name="school_search_form[school_name]"]').type("east{enter}");
+
+    cy.get("h2").contains("14 results found");
+
+    cy.get('[type="checkbox"][name="partnership_form[schools][]"]').as(
+      "checkboxes"
+    );
+
+    cy.get("@checkboxes").eq(0).should("be.checked").click();
+    cy.get("@checkboxes").eq(1).should("not.be.checked").click();
+    cy.get("@checkboxes").eq(2).should("not.be.checked").click();
+    cy.get("@checkboxes").eq(4).should("not.be.checked").click();
+
+    cy.get(".partnerships-submit")
+      .should("contain", "Add partnerships with 7 schools")
+      .should("not.be.disabled");
```

Comment:
This test is incomplete as it isn't testing that the hidden elements are being added - ideally though that test will be done by clicking the add partners button and seeing what happens on the next page, so it's waiting for that functionality first.